src/interval.jl

```diff
@@ -346,3 +346,8 @@ end
 function TimeZones.astimezone(i::Interval{ZonedDateTime}, tz::TimeZone)
     return Interval(astimezone(first(i), tz), astimezone(last(i), tz), inclusivity(i))
 end
+
+function TimeZones.timezone(i::Interval{ZonedDateTime})
+    timezone(first(i)) == timezone(last(i)) && return timezone(first(i))
+    throw(ArgumentError("Interval $i contains mixed timezones."))
```

Comment:
Definitely personal preference but I think this is a more standard way of writing this which I find easier to read:
```suggestion
    if timezone(first(i)) != timezone(last(i))
        throw(ArgumentError("Interval $i contains mixed timezones."))
    end
    return timezone(first(i))
```

Reply:
yeah I guess throwing an error as the last line isn't very safe.